resources/experimental-feature.md

```diff
@@ -0,0 +1,89 @@
+# Experimental features: description and process
+
+## Quick definition of experimental feature
+
+An experimental feature is a feature present in the final Meilisearch binary, but that is not considered as stable: the API can break between 2 versions of Meilisearch.
+
+Even if he feature is present in the Meilisearch binary, it will be not enabled by default: the users must activate it to use it consciously.
+
+> ⚠️ Experimental features are NOT [prototypes]().
+
+## Motivations
+
+From the moment Meilisearch v1 is released, Meilisearch is considered a stable binary where the API cannot break. It will be impossible to provide a feature and break it without releasing a major version.
+
+Since we don’t plan to release a new major version every two months (we guarantee stability to our users), we need to ensure we are making the right product decisions about our newly integrated features.
+
+To prevent this, we want to provide first these features as "experimental" by letting users testing them and reporting any issues they could have with their usage.
+
+## Which features are concerned?
+
+Not all the new feature introduction will get through the experimental feature process.
+
+Experimental features are:
+
+- New features we want to add to the search engine (like a new route, a new parameter, or a new CLI flag), but we are not sure if we want to provide them this way or even at all. We need the user feedback to make our final decision.
+- Improvements we want to provide, but the engine team is not comfortable considering them immediately as stable. For instance, relevancy or performance improvements.
+- Features that would introduce a breaking change and so can’t be integrated as stable before the v2 is released.
+- Features we want to provide into the search engine, but that will NEVER be stable. These features are useful to provide a quick fix to the users when they have critical issues when using Meilisearch, for instance, an option to disable the auto-batching feature.
+
+## How to use experimental features?
+
+They will be integrated into the stable binary but must be activated with a CLI flag.
```

Comment:
In the future, we may be able to activate some at run-time via the API

Reply:
Yes: the guide will have to be upgraded if we change this. Any change in the process should lead to an upgrade of the guide (with a PR so that everyone can see and review)